src/main/scala/Robot.scala

```diff
@@ -8,58 +8,68 @@ object Directions {
   object Right extends Directions
 }
 
-class Robot {
-  var coordinates: (Int, Int) = (0, 0)
+case class Coordinates(x:Int = 0, y:Int = 0)
```

Comment:
Fair enough. This had a smell to me as well, because when you create a zeroed out `Coordinate`, it looks like 
```scala
zeroedCoordinate = new Coordinate()
``` 
which is not very clear on what `Coordinate` is doing, versus
```scala
zeroedCoordinate = new Coordinate(0, 0)
``` 
which is much, much easier to grok.

Updated and pushed.